tests/global.rs

```diff
@@ -134,3 +138,87 @@ fn hammer_split() {
         assert_eq!(N * 4, cnt.load(Relaxed));
     }
 }
+
+#[test]
+fn spawn_handle() {
+    let f = lazy(|| {
+        Ok::<_, ()>("hello from the future")
+    });
+
+    let res = tokio::run(lazy(move || {
+        tokio::spawn_handle(f)
+            .then(|res| {
+                assert_eq!(res.unwrap(), "hello from the future");
+                Ok(())
+            })
+    }));
+
+}
+
+#[test]
+fn spawn_handle_error() {
+    let f = lazy(|| {
+        Err::<(), _>("something is wrong")
+    });
+
+    tokio::run(lazy(move || {
+        tokio::spawn_handle(f)
+            .or_else(|e| {
+                assert_eq!(e.into_inner(), Some("something is wrong"));
```

Comment:
I would use `then` here. There could plausibly be a case where `spawn_handle` completes successfully (w/ a bug).

Also, the panic note from above applies here as well I believe.